src/include/distributed/extended_op_node_utils.h

```diff
@@ -0,0 +1,37 @@
+/*-------------------------------------------------------------------------
+ *
+ * extended_op_node_utils.h
+ *	  General Citus planner code.
+ *
+ * Copyright (c) 2017, Citus Data, Inc.
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef EXTENDED_OP_NODE_UTILS_H_
+#define EXTENDED_OP_NODE_UTILS_H_
+
+#include "distributed/multi_logical_planner.h"
+
+
+/*
+ * ExtendedOpNodeStats is a helper structure that is used to
+ * share the common information among the worker and master extended
+ * op nodes.
+ *
+ * It is designed to be a read-only singleton object per extended op node
+ * generation and processing.
+ */
+typedef struct ExtendedOpNodeStats
+{
+	bool groupedByDisjointPartitionColumn;
+	bool repartitionSubquery;
+	bool hasNonPartitionColumnDistinctAgg;
+	bool pullDistinctColumns;
+	bool pushDownWindowFunctions;
+} ExtendedOpNodeStats;
```

Comment:
Stats suggests this is about monitoring. Maybe Properties or Flags? 

Also, since we only have flags in this struct, I wonder whether this could be an integer.

Reply:
I think I'd prefer to go with the `Properties` given that it's likely we could expand this in the future for non-boolean things as well.